ci/test.py

```diff
@@ -424,11 +452,24 @@ def test_guest_image(self):
         if not self.test_image:
             self.skipTest("%s not defined" % TEST_IMAGE_ENV_VAR)
 
+        # Get the test image name (strip out the flavor identifier) identifier.
+        image_name = self.test_image.rsplit('-', 1)[0]
+
+        expected_dist_id = image_name.split('-', 1)[0]
+        expected_release_str = image_name.split('-', 1)[1]
+
+        # Capture the LSB release information.
         dist_id_str = self.assertPyrexContainerCommand('lsb_release -i', quiet_init=True, capture=True).decode('utf-8').rstrip()
         release_str = self.assertPyrexContainerCommand('lsb_release -r', quiet_init=True, capture=True).decode('utf-8').rstrip()
 
-        self.assertRegex(dist_id_str.lower(), r'^distributor id:\s+' + re.escape(self.test_image.split('-', 1)[0]))
-        self.assertRegex(release_str.lower(), r'^release:\s+' + re.escape(self.test_image.split('-', 1)[1]))
+        self.assertRegex(dist_id_str.lower(), r'^distributor id:\s+' + re.escape(expected_dist_id))
+        self.assertRegex(
+            release_str.lower(),
+            # If we are using the centos 7 image, we need to match the
+            # major[.minor[.micro]] version pattern. we only care if the major
+            # version is 7.
+            r'^release:\s+' + re.escape(expected_release_str) + (r'\.[0-9]+\.[0-9]+' if expected_dist_id == 'centos' else '')
+        )
```

Comment:
The inline conditional here makes this hard to read... also the old test only check that the prefix is present anyway, so I think the following minor tweak to the original would catch all the cases we care about:
```python
self.assertRegex(release_str.lower(), r'^release:\s+' + re.escape(self.test_image.split('-', 1)[1]) + r'[\.$ ]')
```

Reply:
What are the `[]` for in
```python
r'[\.$]'
```
shouldn't it just be:
```python
r'.*$'
```
?

Reply:
I think you want to match any of `.`, whitespace, or end of string following the version number. That means given a `TEST_IMAGE` of `ubuntu-14.04` the following would match:
```
14.04
14.04.1
14.04 foo
```
but any of the following would not match:
```
14.041
14.04a
```

Maybe the space isn't necessary to match on and we can try just `r[\.$]`

Reply:
my approach worked.

Reply:
used `r'([0-9.]+|$)'`